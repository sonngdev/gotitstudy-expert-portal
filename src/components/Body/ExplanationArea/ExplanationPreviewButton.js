import React from 'react';
import PropTypes from 'prop-types';
import { TinyEditor } from './TinyEditor';
import { ButtonWithModal } from '../../shared';

export function ExplanationPreviewButton({ explanations }) {
  return (
    <ButtonWithModal
      buttonProps={{
        className: 'flag-session',
        variant: 'positive',
      }}
      buttonAction="Preview â–¶"
      modalSize="medium"
      modalTitle="Explanation preview"
      modalBody={(
        <TinyEditor
          config={{ toolbar: false, statusbar: false, readonly: false }}
          value={(
            `<h4>Theory or Concept</h4>
            ${explanations[0]}
            <br />
            <h4>Step By Step Instructions</h4>
            ${explanations[1]}
            <br />
            <h4>Final Answer</h4>
            ${explanations[2]}`
          )}
        />
      )}
      confirmButtonText="Submit"
      onConfirm={() => window.location.reload()}
    />
  );
}

ExplanationPreviewButton.propTypes = {
  explanations: PropTypes.arrayOf(PropTypes.string),
};

ExplanationPreviewButton.defaultProps = {
  explanations: ['', '', ''],
};
