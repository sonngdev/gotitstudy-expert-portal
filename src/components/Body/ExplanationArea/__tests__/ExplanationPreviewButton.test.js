import React from 'react';
import { shallow } from 'enzyme';
import { ExplanationPreviewButton } from '../ExplanationPreviewButton';

describe('<ExplanationPreviewButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ExplanationPreviewButton
        explanations={[
          '<p>This is theory</p>',
          '<img src="/graph" />',
          '<span>Answer is 3</span>',
        ]}
      />,
    );
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('displays TinyEditor in modal', () => {
    const modal = wrapper.dive().find('Modal');
    expect(modal.find('TinyEditor').exists()).toBe(true);
  });

  it('displays the explanations of all 3 sections', () => {
    const value = wrapper.dive().find('TinyEditor').prop('value');

    expect(value).toContain('<h4>Theory or Concept</h4><p>This is theory</p>');
    expect(value).toContain('<h4>Step By Step Instructions</h4><img src="/graph" />');
    expect(value).toContain('<h4>Final Answer</h4><span>Answer is 3</span>');
  });

  it('reloads window when modal is confirmed', () => {
    mockWindowReload(() => {
      wrapper.dive().find('Modal').find('.confirm').simulate('click');
      expect(window.location.reload).toHaveBeenCalled();
    });
  });
});
