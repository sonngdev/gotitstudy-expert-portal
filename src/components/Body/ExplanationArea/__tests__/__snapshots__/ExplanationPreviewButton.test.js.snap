// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExplanationPreviewButton /> renders 1`] = `"<ButtonWithModal buttonProps={{...}} buttonAction=\\"Preview â–¶\\" modalSize=\\"medium\\" modalTitle=\\"Explanation preview\\" modalBody={{...}} confirmButtonText=\\"Submit\\" onConfirm={[Function: onConfirm]} />"`;
